test-basic/documents-negative.js

```diff
@@ -112,12 +112,16 @@ describe('document negative', function(){
     }, function(error){
 // TODO: REMOVE TEMPORARY WORKAROUND BELOW
 //      error.statusCode.should.equal(400);
-      error.should.have.property('body');
-      error.body.should.have.property('errorResponse');
-      error.body.errorResponse.should.have.property('message');
-      error.body.errorResponse.message.should.containEql('XDMP-RWINVAL');
-      error.body.errorResponse.message.should.containEql('set-transaction');
-      done();
+      try {
+        error.should.have.property('body');
+        error.body.should.have.property('errorResponse');
+        error.body.errorResponse.should.have.property('message');
+        error.body.errorResponse.message.should.containEql('XDMP-RWINVAL');
+        error.body.errorResponse.message.should.containEql('set-transaction');
+        done();
+      } catch(error){
+        done(error);
```

Comment:
Does this result in the test failing - i.e. the error is caught, and then `done(error)` results in a failed test with the message from `error`?

Reply:
This will catch any assertions errors thrown from line 116-120. Without the try-catch block, if there were any assertion failures, the done would not be called and the test-run would show timeout. This way, we can see the actual error.